src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/RazorSemanticTokensEndpoint.cs

```diff
@@ -85,13 +93,13 @@ public async Task<SemanticTokens> Handle(SemanticTokensRangeParams request, Canc
                 throw new ArgumentNullException(nameof(request));
             }
 
-            var documentSnapshot = await TryGetCodeDocumentAsync(request.TextDocument.Uri.GetAbsolutePath(), cancellationToken);
-            if (documentSnapshot is null)
+            var (documentSnapshot, documentVersion) = await TryGetCodeDocumentAsync(request.TextDocument.Uri.GetAbsolutePath(), cancellationToken);
```

Comment:
Feels slightly odd to bake this into `TryGetDocumentAsync` is that only because it has to jump to the foreground thread? If that's the case then i'd suggest renaming the method to `TryGetDocumentInfo`. Or possibly better would be to bake the document version retrieval into the semantic tokens service since it's conceptually probably better there. It's conceptually better there because the version is resolvable via the `documentSnapshot` so it doesn't make a ton of sense to pass in the extra information + what happens if you choose to never send the request, the data would be unneeded. What I'm trying to say is that the semantic token service is the one that cares about the version not this endpoint😄